charts/substra-backend/templates/deployment-test-pod.yaml

```diff
@@ -0,0 +1,98 @@
+{{- if .Values.testPod.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "substra.fullname" . }}-test-pod
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app.kubernetes.io/name: {{ template "substra.name" . }}-test-pod
+    app.kubernetes.io/part-of: {{ template "substra.name" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+        app.kubernetes.io/name: {{ template "substra.name" . }}-test-pod
+        app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ template "substra.name" . }}-test-pod
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      {{- with $.Values.backend.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range . }}
+        - name: {{ . }}
+      {{- end }}
+      {{- end }}
+      containers:
+      - name: test-pod
+        image: "{{ .Values.testPod.image.repository }}:{{ .Values.testPod.image.tag }}"
+        {{- if .Values.backend.image.pullPolicy }}
+        imagePullPolicy: "{{ .Values.backend.image.pullPolicy }}"
+        {{- end }}
+        command: ["/bin/bash", "-c"]
+        args:
+          - |
+            cd substra-tests
+            make test
+            while [ true ];
+                sleep 10;
+            done
+        env:
+          - name: PYTHONUNBUFFERED
+            value: "1"
+        {{- with .Values.extraEnv }}
+{{ toYaml . | indent 10 }}
+        {{- end }}
+        volumeMounts:
+          - name: config
+            mountPath: /conf/{{ .Values.organization.name }}/substra-backend
+            readOnly: true
+        resources:
+          {{- toYaml .Values.testPod.resources | nindent 12 }}
+      initContainers:
+      - name: prepare-base
+        image: "{{ .Values.testPod.image.repository }}:{{ .Values.testPod.image.tag }}"
+        command: ["/bin/bash", "-c"]
+        args:
+          - |
+            git clone https://github.com/SubstraFoundation/substra
+            git clone git@github.com:SubstraFoundation/substra-tests.git
+            cd substra-tests
+            pip install -r requirements.txt
+            pip install -e ../substra
+
+      {{- if .Values.testPod.extraRepos }}
+      - name: prepare-extras
+        image: "{{ .Values.testPod.image.repository }}:{{ .Values.testPod.image.tag }}"
+        command: ["/bin/bash", "-c"]
+        args:
+          - |
+            {{- with $.Values.testPod.extraRepos }}
+            {{- range . }}
+                git clone {{ .url }}
+                cd {{ .name }}
+                [ -f requirements.txt ] && pip install -r requirements.txt
+            {{- end }}
+            {{- end }}
+      {{- end }}
```

Comment:
You will loose those files if you don't use a volume shared between all the pods. You can use a simple emptyDir.